src/app/oauth2/middleware.js

```diff
@@ -29,6 +29,7 @@ module.exports = {
 
       req.session.shared_claims = shared_claims;
       req.session.passportSessionId = passportSessionId;
+      req.session.redirect_url = apiResponse.data.redirect_uri;
```

Comment:
On line 28 can we not do
let { passportSessionId, shared_claims, redirect_uri } = apiResponse.data;

      req.session.redirect_uri = redirect_uri;

Keeps consistency and saves mapping back and forth between uri and url